packages/core/src/ObservableActorRef.ts

```diff
@@ -1,41 +1,47 @@
 import { EventObject, ActorRef, Behavior, SCXML, ActorContext } from './types';
-import { Actor } from './actor';
 import { symbolObservable, toSCXMLEvent } from './utils';
+import { Mailbox } from './Mailbox';
+import { LifecycleSignal, startSignal, stopSignal } from './actors';
+
+const nullSubscription = {
+  unsubscribe: () => void 0
+};
 
 export class ObservableActorRef<TEvent extends EventObject, TEmitted>
   implements ActorRef<TEvent, TEmitted> {
   private current: TEmitted;
   public deferred = true;
   private context: ActorContext<TEvent, TEmitted>;
-  private actor: Actor<TEvent, TEmitted>;
   public name: string;
+  private mailbox = new Mailbox(this._process.bind(this));
 
-  constructor(behavior: Behavior<TEvent, TEmitted>, name: string) {
+  constructor(public behavior: Behavior<TEvent, TEmitted>, name: string) {
     this.name = name;
     this.context = {
       self: this,
       name: this.name,
       observers: new Set(),
       _event: toSCXMLEvent({ type: 'xstate.init' }) as SCXML.Event<TEvent> // TODO: fix
     };
-    this.actor = new Actor(behavior, name, this.context);
-    this.current = this.actor.current;
+    this.current = this.behavior.initialState;
   }
   public start() {
     this.deferred = false;
-    this.actor.start();
+    this.mailbox.enqueue(startSignal);
+    this.mailbox.start();
 
     return this;
   }
   public stop() {
-    this.actor.stop();
+    this.mailbox.clear();
+    this.mailbox.enqueue(stopSignal);
     return this;
   }
   public subscribe(observer) {
-    return this.actor.subscribe(observer);
+    return this.behavior.subscribe?.(observer) ?? nullSubscription;
```

Comment:
Also, a potential refactor item - but I don't fully see why a behavior would have a `subscribe` method (especially that a behavior, at least in Akka terms, should be pure and `subscribe` requires keeping an internal state).

Note that even if `behavior` would have a `subscribe` method we can't truly just proxy to it here. This exact implementation made me to skip some `@xstate/react` tests because subscribers are called from within a behavior before we have a chance to assign the updated `this.current` in this very class here. This leads to "stale" notifications being emitted

Reply:
Will address this in a separate PR (and remove `behavior.subscribe`).

The tentative plan is to expose `actorContext.emit(data)` since the emitted data isn't necessarily tied to the `state`, which may be private for certain use-cases.